modules/lang-painless/src/main/java/org/elasticsearch/painless/node/ECapturingFunctionRef.java

```diff
@@ -88,6 +88,7 @@ void analyze(Locals variables) {
     @Override
     void write(MethodWriter writer, Globals globals) {
         writer.writeDebugInfo(location);
+        checkWriteBranch(null);
```

Comment:
For branching during the write phase, AExpression nodes contain Label tru and Label fals as variables -- these are passed to child nodes as necessary and used to do a jump in child node in certain cases (afaik this mimics exactly what javac generates as byte code).  Certain nodes should never have one of these variables be non-null, and while this should be caught under normal circumstances during analysis, if theres's a bug I don't want to throw a stack overflow error or verify error as the user.  checkWriteBranch checks a number of conditions and will throw an error if necessary.  When writer is passed in as null it also checks the condition (try != null || fals != null) and will throw an exception if either one is non-null.